packages/eslint-config-nimble/typescript.js

```diff
@@ -96,96 +44,14 @@ module.exports = {
                 'import/no-extraneous-dependencies': [
                     'error',
                     { devDependencies: true }
-                ]
-            }
-        },
-        {
-            files: ['*.foundation.spec.ts'],
-            rules: {
-                'no-restricted-imports': [
-                    'error',
-                    { paths: restrictedImportsPaths() }
-                ]
-            }
-        },
-        {
-            files: ['styles.ts'],
-            rules: {
-                // Prettier and eslint conflict in how they format CSS in styles files and we prefer prettier's output
-                '@typescript-eslint/indent': 'off'
-            }
-        },
-        {
-            files: ['template.ts'],
-            rules: {
-                // Using '??' in templates does not get flagged correctly by FAST as being a volatile binding.
-                // See https://github.com/ni/nimble/issues/1843 for more information.
-                'no-restricted-syntax': [
-                    'error',
-                    { selector: "LogicalExpression[operator='??']" }
-                ]
-            }
-        },
-        {
-            // Instead of enums, this repo uses const objects and type unions which should live in types.ts
-            files: ['types.ts'],
-            rules: {
-                // The const object and type union should have the same name
-                '@typescript-eslint/no-redeclare': 'off',
-                // Enum-like variables should use PascalCase and values should use camelCase
-                // Also allow camelCase variables for things like attribute strings
-                '@typescript-eslint/naming-convention': [
-                    'error',
-                    {
-                        selector: 'objectLiteralProperty',
-                        format: ['camelCase']
-                    },
-                    {
-                        selector: 'variable',
-                        format: ['camelCase', 'PascalCase']
-                    },
-                    {
-                        selector: 'typeLike',
-                        format: ['PascalCase']
-                    },
-                    {
-                        selector: 'default',
-                        format: ['camelCase'],
-                        leadingUnderscore: 'allow',
-                        trailingUnderscore: 'allow'
-                    }
-                ]
-            }
-        },
-        {
-            files: ['**/*.react.tsx'],
-            rules: {
-                // The React components should use PascalCase
-                '@typescript-eslint/naming-convention': [
-                    'error',
-                    {
-                        selector: 'objectLiteralProperty',
-                        format: ['PascalCase']
-                    }
-                ]
+                ],
+
+                'max-classes-per-file': 'off'
```

Comment:
nit: Always nice to have an explanation for rule customizations
```suggestion
                // It's common to define helper classes for tests and it's more readable to do this in the same file
                'max-classes-per-file': 'off'
```